RecoHI/HiJetAlgos/plugins/HiPuRhoProducer.cc

```diff
@@ -0,0 +1,584 @@
+    inputs_.push_back(&input);
+
+  fjInputs_.reserve(inputs_.size());
+  inputTowers();
+  fjOriginalInputs_ = fjInputs_;
+  setInitialValue_ = true;
+  calculatePedestal(fjInputs_);
+  subtractPedestal(fjInputs_);
+
+  fjJetDefinition_ = std::make_shared<fastjet::JetDefinition>(fastjet::antikt_algorithm, rParam_);
+  fjClusterSeq_ = std::make_shared<fastjet::ClusterSequence>(fjInputs_, *fjJetDefinition_);
+  fjJets_ = fastjet::sorted_by_pt(fjClusterSeq_->inclusive_jets(puPtMin_));
+
+  etaEdgeLow_.clear();
+  etaEdgeHi_.clear();
+  etaEdges_.clear();
+
+  rho_.clear();
+  rhoExtra_.clear();
+  rhoM_.clear();
+  nTow_.clear();
+
+  towExcludePt_.clear();
+  towExcludePhi_.clear();
+  towExcludeEta_.clear();
+
+  setInitialValue_ = false;
+  std::vector<fastjet::PseudoJet> orphanInput;
+  calculateOrphanInput(orphanInput);
+  calculatePedestal(orphanInput);
+  putRho(iEvent, iSetup);
+
+  inputs_.clear();
+  fjInputs_.clear();
+  fjJets_.clear();
+}
+
+void HiPuRhoProducer::inputTowers() {
+  int index = -1;
+  for (auto const& input : inputs_) {
+    index++;
+
+    if (edm::isNotFinite(input->pt()))
+      continue;
+    if (input->pt() < 100 * std::numeric_limits<double>::epsilon())
+      continue;
+
+    fjInputs_.push_back(fastjet::PseudoJet(input->px(), input->py(), input->pz(), input->energy()));
+    fjInputs_.back().set_user_index(index);
+  }
+}
+
+void HiPuRhoProducer::setupGeometryMap(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  LogDebug("PileUpSubtractor") << "The subtractor setting up geometry...\n";
+  const auto& pG = iSetup.getData(caloGeometryToken_);
+  geo_ = &pG;
+  std::vector<DetId> alldid = geo_->getValidDetIds();
+  int ietaold = -10000;
+  ietamax_ = -10000;
+  ietamin_ = 10000;
+  towermap_.clear();
+
+  for (auto const& did : alldid) {
+    if (did.det() == DetId::Hcal) {
+      HcalDetId hid = HcalDetId(did);
+      allgeomid_.push_back(did);
+      towermap_.push_back({hid.ieta(), hid.iphi(), geo_->getPosition(did).eta(), geo_->getPosition(did).phi()});
+      if (hid.ieta() != ietaold) {
+        ietaold = hid.ieta();
+        geomtowers_[hid.ieta()] = 1;
+        if (hid.ieta() > ietamax_)
+          ietamax_ = hid.ieta();
+        if (hid.ieta() < ietamin_)
+          ietamin_ = hid.ieta();
+      } else {
+        geomtowers_[hid.ieta()]++;
+      }
+    }
+  }
+
+  for (auto const& gt : geomtowers_) {
+    int it = gt.first;
+    int vi = it - 1;
+
+    if (it < 0)
+      vi = nEtaTow_ + it;
+
+    if (vi < nEtaTow_)
+      vngeom_[vi] = gt.second;
+  }
+}
+
+void HiPuRhoProducer::calculatePedestal(std::vector<fastjet::PseudoJet> const& coll) {
+  LogDebug("PileUpSubtractor") << "The subtractor calculating pedestals...\n";
+
+  std::map<int, double> emean2;
+  std::map<int, int> ntowers;
+
+  int ietaold = -10000;
+  // Initial values for emean_, emean2, esigma_, ntowers
+
+  for (int vi = 0; vi < nEtaTow_; ++vi) {
+    int it = vi + 1;
+    if (it > nEtaTow_ / 2)
+      it = vi - nEtaTow_;
```

Comment:
```suggestion
```
to resolve the issue reported in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-561667/8025/llvm-analysis/report-a044d2.html#EndPath

that's unless the change made in https://github.com/cms-sw/cmssw/pull/30057/files/a0e15386a5d2675bc8eeba5558d4bc905cfd65d0..e8657815f4596a3349308006bf8981f9d004ca6b#diff-2093a69ad726653693a30f6c4415e430L308 (click "Load diff" for RecoHI/HiJetAlgos/plugins/HiPuRhoProducer.cc to get to the change), which had dependence on ```it``` was made incorrectly.

This appears to be the last comment for this PR. Sorry for missing it in the last round(s).